argoverse/utils/se3.py

```diff
@@ -27,16 +27,13 @@ def transform_point_cloud(self, point_cloud: np.ndarray) -> np.ndarray:
         """Apply the SE(3) transformation to this point cloud.
 
         Args:
-            point_cloud: Array of shape (N, 3)
+            point_cloud: Array of shape (N, 3). If the transform represents dst_SE3_src,
+                then point_cloud should consist of points in frame `src`
 
         Returns:
-            transformed_point_cloud: Array of shape (N, 3)
+            Array of shape (N, 3) representing the transformed point cloud, i.e. points in frame `dst`
         """
-        # convert to homogeneous
-        num_pts = point_cloud.shape[0]
-        homogeneous_pts = np.hstack([point_cloud, np.ones((num_pts, 1))])
-        transformed_point_cloud = homogeneous_pts.dot(self.transform_matrix.T)
-        return transformed_point_cloud[:, :3]
+        return point_cloud @ self.rotation.T + self.translation
 
     def inverse_transform_point_cloud(self, point_cloud: np.ndarray) -> np.ndarray:
         """Undo the translation and then the rotation (Inverse SE(3) transformation)."""
```

Comment:
Could we also edit this function accordingly so that we can do the operation in one line.

Reply:
Thanks for the review Tanmay -- I'm not sure this would be more readable on one line, since it would look like
```
    def inverse_transform_point_cloud(self, point_cloud: np.ndarray) -> np.ndarray:
        """Undo the translation and then the rotation (Inverse SE(3) transformation)."""
        return (point_cloud.copy() - self.translation) @ self.rotation
```
@benjaminrwilson what do you think?

Reply:
I prefer the one-liner because it makes it easier for me to read it as an equation. On a separate note, do you think we should have an option to transform `point_cloud` in-place? What do you think is the most intuitive design decision when comparing expected behavior to speed?